wrapanapi/systems/virtualcenter.py

```diff
@@ -241,14 +286,166 @@ def _get_resource_pool(self, resource_pool_name=None):
+                                                vm_reloc_spec=vm_reloc_spec)
+
+        # Get the location of the new VM
+        folder = self._get_loc_of_vm(source_template=source_template,
+                                     progress_callback=progress_callback)
+
+        action = source_template.RelocateVM_Task if relocate else source_template.CloneVM_Task
+        action_args = dict(spec=vm_reloc_spec) if relocate else dict(folder=folder,
+                                                                     name=destination,
+                                                                     spec=vm_clone_spec)
+
+        return action(**action_args)
+
+    def _clone_on_datastore_cluster(self, destination, resourcepool, datastore, power_on, sparse,
+                                  template, progress_callback, cpu, ram, host,
+                                  source_template, deploy_on_ds_cluster):
+        """Set all required parameters for a clone or relocate on a datastore cluster"""
+
+        vm_reloc_spec = self._set_vm_relocate_spec(resourcepool=resourcepool, host=host,
+                                                   sparse=sparse,
+                                                   progress_callback=progress_callback,
+                                                   deploy_on_ds_cluster=deploy_on_ds_cluster)
+
+        vm_clone_spec = self._set_vm_clone_spec(power_on=power_on, template=template,
+                                                cpu=cpu, ram=ram,
+                                                vm_reloc_spec=vm_reloc_spec)
+
+        # Create the StoragePlaceSpec object that will be passed to the RecommendDatastores method
+        storage_spec = vim.StoragePlacementSpec()
+        # Create a empty Config spec data object
+        config = vim.vm.ConfigSpec()
+        # Specification for moving or copying a VM to a different storage pod
+        pod_spec = vim.StorageDrsPodSelectionSpec(storagePod=datastore)
+
+        storage_spec.type = 'clone'
+        storage_spec.cloneName = destination
+        storage_spec.folder = self._get_loc_of_vm(source_template=source_template,
+                                                  progress_callback=progress_callback)
+        storage_spec.podSelectionSpec = pod_spec
+        storage_spec.vm = source_template
+        storage_spec.cloneSpec = vm_clone_spec
+        storage_spec.configSpec = config
+
+        # Returns a StoragePlacementResult object.
+        # For SDRS-enabled pods, this API is intended to replace RelocateVM_Task and CloneVM_Task
+        # SDRS is required for managing aggregated resources of a datastore cluster
+        result = self.system.content.storageResourceManager.RecommendDatastores(
+            storageSpec=storage_spec)
+
+        if result:
+            key = result.recommendations[0].key
+        else:
+            raise Exception(
+                "RecommendDatastore task failed to provide host for  {}".format(destination))
```

Comment:
two spaces

Reply:
and I'd rather raise non generic exception in this case

Reply:
I guess AttributeError won't suit in this case. I'd use ValueError or some custom exception